firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ImageUtils.java

```diff
@@ -14,51 +14,78 @@
 
 package com.google.firebase.appdistribution.impl;
 
+import android.content.ContentResolver;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
-import androidx.annotation.Nullable;
+import android.net.Uri;
 import com.google.auto.value.AutoValue;
-import java.io.File;
+import com.google.firebase.appdistribution.FirebaseAppDistributionException;
+import com.google.firebase.appdistribution.FirebaseAppDistributionException.Status;
+import java.io.IOException;
+import java.io.InputStream;
 
 public class ImageUtils {
 
+  private static final String TAG = "ImageUtils";
+  public static final int MAX_IMAGE_READ_RETRIES = 10;
+  public static final int IMAGE_READ_RETRY_SLEEP_MS = 300;
+
   @AutoValue
   abstract static class ImageSize {
     abstract int width();
 
     abstract int height();
 
-    static ImageSize read(File file) {
+    static ImageSize read(InputStream inputStream) {
       final BitmapFactory.Options options = new BitmapFactory.Options();
       options.inJustDecodeBounds = true;
-      BitmapFactory.decodeFile(file.getAbsolutePath(), options);
+      BitmapFactory.decodeStream(inputStream, /* outPadding= */ null, options);
       return new AutoValue_ImageUtils_ImageSize(options.outWidth, options.outHeight);
     }
   }
 
   /**
-   * Read an image from a file, scaled as small as possible according to the target size.
+   * Read an image, scaled as small as possible according to the target size.
    *
    * <p>The returned bitmap will be scaled down, preserving the aspect ratio, by the largest power
    * of 2 that results in the width and height still being larger than the target.
    *
    * <p>Based on https://developer.android.com/topic/performance/graphics/load-bitmap#load-bitmap.
    *
-   * @return the image, or null if it could not be decoded
-   * @throws IllegalArgumentException if target height or width are less than or equal to zero
+   * @return the image
+   * @throws FirebaseAppDistributionException if the image could not be read
    */
-  @Nullable
-  public static Bitmap readScaledImage(File file, int targetWidth, int targetHeight) {
+  public static Bitmap readScaledImage(
+      ContentResolver contentResolver, Uri uri, int targetWidth, int targetHeight)
+      throws FirebaseAppDistributionException {
     if (targetWidth <= 0 || targetHeight <= 0) {
-      throw new IllegalArgumentException(
+      throw new FirebaseAppDistributionException(
```

Comment:
I think this can remain an `IllegalArgumentException` - it's clearly a programmer error if negative (or zero) values are passed in. I don't think we want to even catch the problem.